openwhisk/actionProxy.go

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package openwhisk
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+)
+
+// theServer is the current server
+var theServer http.Server
+
+// theChannel is the channel communicating with the action
+var theChannel chan string
+
+// theChannel is the channel communicating to flush the logs
+var theLogger chan bool
+
+func stopAction() {
+	// terminate current action
+	if theChannel != nil {
+		log.Println("terminating old action")
+		theChannel <- ""
+		theChannel = nil
```

Comment:
Should the channel be closed instead of assigning it to nil? This will automatically unblock all listeners if any

Reply:
actually I already did a BIG refactoring of the pipe_exec that is greatly simplifying those cases. It lives here https://github.com/sciabarracom/incubator-openwhisk-runtime-go/blob/new-executor/openwhisk/executor.go
I have not yet merged with the master. Currently the PR works with the cases listed in the test/test.t that covers all the use cases discussed with Rodric.

Reply:
Maybe it would make sense to merge the refactoring into this PR? That would make it easier to review it, to avoid commenting things that were already reworked...

Reply:
Yes but I applied your comments to my refactoring. I have almost finished it.